### ISSUE TITLE
How to run git submodule update --init within Docker container

### ISSUE THREAD

**Issue by taut-and-yare** [yourlabs/django-autocomplete-light#1040]
I am using dal with Django 1.11. My deployment environment is Divio cloud.

However, I hadn't pinned it down to my requirements.in file, therefore when I tried to rebuild the project it failed due to a reference of the new version (>3.3.0-rc6):
`from django.contrib.admin.widgets import SELECT_2_TRANSLATIONS`

When I try to use a previous version adding into my requirements.in the following: 
`https://github.com/yourlabs/django-autocomplete-light/archive/3.3.0-rc6.zip#egg=django-autocomplete-light`

... I get an error due to _submodules not included automatically for archive versions_ as also mentioned in the [docs](https://django-autocomplete-light.readthedocs.io/en/master/install.html).

My deployment environment means I am working within a Docker container. What would be the correct way of dealing with this? i.e. how to run git submodule update --init within the Docker container? I failed doing it via the bash:
`root@xxxxx:/app# git -C /virtualenv/lib/python2.7/site-packages/dal submodule update --init
fatal: Not a git repository (or any of the parent directories): .git`

I had to resort to locally cloning dal, then checking out version 3.3.0-rc6, then running `git submodule update --init` and then copying this into my own project's django apps. It works, but it's not pretty.

**Comment by jpic** [comment]
Did you try pip install -e instead of copying files to your project ?

**Comment by taut-and-yare** [comment]
Hi @jpic ,
I could not explicitly do pip install -e due to working with [pip-tools](https://github.com/jazzband/pip-tools). I can only specify dependencies inside a `requirements.in` file.

**Comment by jpic** [comment]
I don't think 3.3.0 is really compatible with Django 1.11 but pull requests are welcome.

3.3.0 relies on django.contrib.admin vendorizing of select2 in django 2

**Comment by taut-and-yare** [comment]
Indeed, that is why I am trying to roll back to tag 3.3.0-rc6 which still works for django 1.11

**Comment by jpic** [comment]
So, did you try with a regular pip install of 3.3.0-rc6  ?
